source/isaaclab/isaaclab/sensors/tacsl_sensor/visuotactile_sensor.py

```diff
@@ -0,0 +1,990 @@
+        # unsubscribe from callbacks
+        super().__del__()
+
+    def __str__(self) -> str:
+        """Returns: A string containing information about the instance."""
+        return (
+            f"Tactile sensor @ '{self.cfg.prim_path}': \n"
+            f"\tsensor type       : {self.cfg.sensor_type}\n"
+            f"\tupdate period (s) : {self.cfg.update_period}\n"
+            f"\tcamera enabled    : {self.cfg.enable_camera_tactile}\n"
+            f"\tforce field enabled: {self.cfg.enable_force_field}\n"
+            f"\tnum instances     : {self.num_instances}\n"
+        )
+
+    """
+    Properties
+    """
+
+    @property
+    def num_instances(self) -> int:
+        return self._num_envs
+
+    @property
+    def data(self) -> VisuoTactileSensorData:
+        # Update sensors if needed
+        self._update_outdated_buffers()
+        # Return the data
+        return self._data
+
+    """
+    Operations
+    """
+
+    def reset(self, env_ids: Sequence[int] | None = None):
+        """Resets the sensor internals."""
+        # reset the timestamps
+        super().reset(env_ids)
+        self.reset_timing_statistics()
+
+        # Reset camera sensor if enabled
+        if self._camera_sensor is not None:
+            self._camera_sensor.reset(env_ids)
+
+    """
+    Implementation
+    """
+
+    def _initialize_impl(self):
+        """Initializes the sensor-related handles and internal buffers."""
+        super()._initialize_impl()
+
+        # Obtain global simulation view
+        self._physics_sim_view = SimulationManager.get_physics_sim_view()
+
+        # Initialize camera-based tactile sensing
+        if self.cfg.enable_camera_tactile and self.cfg.camera_cfg is not None:
+            self._initialize_camera_tactile()
+
+        # Initialize force field tactile sensing
+        if self.cfg.enable_force_field:
+            self._initialize_force_field()
+
+        # Initialize visualization
+        if self.cfg.debug_vis:
+            self._initialize_visualization()
+
+    def setup_compliant_materials(self):
+        """Setup compliant contact materials for the elastomer.
+
+        This method configures the elastomer collision geometry with compliant contact
+        materials that provide softer, more realistic contact behavior. Only applies
+        materials if compliant contact is enabled in the configuration.
+
+        Note:
+            This method should be called after sensor initialization and before
+            simulation starts to ensure proper contact behavior.
+        """
+        # Get the current stage
+        stage = stage_utils.get_current_stage()
+
+        # Create material configuration
+        print("set up compliant contact materials", self.cfg.elastomer_collision_path)
+        material_cfg = RigidBodyMaterialCfg(
+            compliant_contact_stiffness=self.cfg.compliance_stiffness,
+            compliant_contact_damping=self.cfg.compliant_damping,
+        )
```

Comment:
I think it is neater to have the user specify the rigid body material configuration than individual parameters. They can then also modify additional params depending on how compliant contact model develops. Right now all these are exposed "one" level up, bloating the main config.